js/privatebin.js

```diff
@@ -2295,6 +2129,192 @@ jQuery.PrivateBin = function($, sjcl, Base64, RawDeflate) {
+                    this.showImagePreview(targetElement, data);
+                } else if (mimeType.match(/video\//i)) {
+                    this.showVideoPreview(targetElement, data, mimeType);
+                } else if (mimeType.match(/audio\//i)) {
+                    this.showAudioPreview(targetElement, data, mimeType);
+                } else if (mimeType.match(/\/pdf/i)) {
+                    this.showPDFPreview(targetElement, data);
+                }
+                //else {
+                //console.log("file but no image/video/audio/pdf");
+                //}
+            }
+        };
+
+        /**
+         * Get Mime Type from a DataURL
+         *
+         * @param {type} dataURL
+         * @returns Mime Type from a dataURL as obtained for a file using the FileReader API https://developer.mozilla.org/en-US/docs/Web/API/FileReader#readAsDataURL()
+         */
+        me.getMimeTypeFromDataURL = function (dataURL) {
+            return dataURL.slice(dataURL.indexOf('data:') + 5, dataURL.indexOf(';base64,'));
+        };
+
+        me.showImagePreview = function (targetElement, image) {
+            targetElement.html(
+                    $(document.createElement('img'))
+                    .attr('src', image)
+                    .attr('class', 'img-thumbnail')
+                    );
+            targetElement.removeClass('hidden');
+        };
+
+        me.showVideoPreview = function (targetElement, video, mimeType) {
+            var videoPlayer = $(document.createElement('video'))
+                    .attr('controls', 'true')
+                    .attr('autoplay', 'true')
+                    .attr('loop', 'true')
+                    .attr('class', 'img-thumbnail');
+
+            videoPlayer.append($(document.createElement('source'))
+                    .attr('type', mimeType)
+                    .attr('src', video));
+            targetElement.html(videoPlayer);
+            targetElement.removeClass('hidden');
+        };
+
+        me.showAudioPreview = function (targetElement, audio, mimeType) {
+            var audioPlayer = $(document.createElement('audio'))
+                    .attr('controls', 'true')
+                    .attr('autoplay', 'true');
+
+            audioPlayer.append($(document.createElement('source'))
+                    .attr('type', mimeType)
+                    .attr('src', audio));
+            targetElement.html(audioPlayer);
+            targetElement.removeClass('hidden');
+        };
+
+        me.showPDFPreview = function (targetElement, pdf) {
+            //PDFs are only displayed if the filesize is smaller than about 1MB (after base64 encoding).
+            //Bigger filesizes currently cause crashes in various browsers.
+            //See also: https://code.google.com/p/chromium/issues/detail?id=69227
+
+            //Firefox crashes with files that are about 1.5MB
+            //The performance with 1MB files is bareable
+            if (pdf.length < 1398488) {
```

Comment:
Maybe make an early return (length > … --> return) instead of this big indentation. It just looks better…